src/main/java/org/trigon/hardware/misc/leds/CANdleLEDStrip.java

```diff
@@ -0,0 +1,153 @@
+package org.trigon.hardware.misc.leds;
+
+import com.ctre.phoenix.led.*;
+import edu.wpi.first.wpilibj.util.Color;
+import org.trigon.hardware.RobotHardwareStats;
+
+import java.util.function.Supplier;
+
+/**
+ * A LED strip that is controlled by a CANdle, and uses AddressableLED for simulation.
+ */
+public class CANdleLEDStrip extends LEDStrip {
+    private static CANdle CANDLE;
+    private static int LAST_CREATED_LED_STRIP_ANIMATION_SLOT = 0;
+    private final int animationSlot;
+
+    /**
+     * Sets the CANdle instance to be used for controlling the LED strips. Must be set before using any LED strips. Should only be called once
+     *
+     * @param candle the CANdle instance to be used
+     */
+    public static void setCANdle(CANdle candle) {
+        if (CANDLE == null)
+            CANDLE = candle;
+    }
+
+    public static void setTotalAmountOfLEDs(int totalAmountOfLEDs) {
+        if (RobotHardwareStats.isSimulation() || RobotHardwareStats.isReplay())
+            AddressableLEDStrip.initiateAddressableLED(0, totalAmountOfLEDs);
+    }
```

Comment:
when is this used and why? candle should have no sim logic

Reply:
This is so that the simulation will work, it should be called in a static block in the constants class. I have this because othewise we would have to call the method in AddressableLED and I think it's more intuative, that it would be in the CANdleLEDStrip class

Reply:
I see, document this better.